### PR TITLE
[ONNX] Support more sympy operations in fx-onnx exporter

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112758

Fix https://github.com/microsoft/onnx-converters-private/issues/190

This PR retires built-in function mapping by adding built-in ops into torchlib (https://github.com/microsoft/onnxscript/pull/1135), and provide a runtime tests to guard the operation conversion.

More built-in ops are supported in torchlib as well.

~~NOTE: `native_batch_norm` regression is caused by https://github.com/microsoft/onnxscript/issues/1140. Will fix it before I merge this.~~

### CODE REVIEW

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 1]
```diff
@@ -129,6 +131,59 @@ def func(x, b=torch.tensor(1.0)):
             func, (tensor_x,), input_kwargs={"b": torch.tensor(5.0)}
         )
 
+    @pytorch_test_common.skip_dynamic_fx_test(
+        "sympy operation tests doesn't need dynamic shape"
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:135]:
```suggestion
        "sympy operation tests don't need dynamic shape"
```
NIT: SPELLING

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 2]
```diff
@@ -129,6 +131,59 @@ def func(x, b=torch.tensor(1.0)):
...
+        self.run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
+            func,
+            (
+                x,
+                y,
+            ),
+        )
+
+    @pytorch_test_common.skip_dynamic_fx_test(
+        "sympy operation tests doesn't need dynamic shape"
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:164]:
```suggestion
        "sympy operation tests don't need dynamic shape"
```

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 3]
```diff
@@ -129,6 +131,59 @@ def func(x, b=torch.tensor(1.0)):
...
+        def func(x, y):
+            return (
+                # torch.tensor([operator.add(x.item(), y.item())]),
+                # torch.tensor([operator.sub(x.item(), y.item())]),
+                # torch.tensor([operator.mul(x.item(), y.item())]),
+                # torch.tensor([operator.truediv(x.item(), y.item())]),
+                # torch.tensor([operator.floordiv(x.item(), y.item())]),
+                # torch.tensor([operator.pow(x.item(), y.item())]),
+                # torch.tensor([operator.abs(x.item())]),
+                # torch.tensor([operator.neg(x.item())]),
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:147]: why all these comments? are they from previous testing?

either delete them or add an additional explanation for them

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py [hunk 4]
```diff
@@ -240,12 +243,17 @@ def _fill_tensor_shape_type(
...
+        elif fx_type_utils.is_torch_symbolic_type(expected_value):
             # aten::sym_size output is a int, not a tensor, which stands
             # for the size of one dim. We treat it as 0-D tensor.
-            onnxscript_value.dtype = fx_type_utils.from_sym_value_to_torch_dtype(
-                expected_value
-            )
+            if node.target == math.ceil or node.target == math.floor:
+                # Type promotion for ceil/floor from built-in functions.
+                # ceil and floor have float outputs in torchlib
+                onnxscript_value.dtype = torch.float32
```
Comment (thiagocrepaldi) [inline torch/onnx/_internal/fx/fx_onnx_interpreter.py:252]: should this be part of the type promotion logic? adding hard-coded op-specific logic in the dispatcher makes it hard to maintain

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/_internal/fx/registration.py [hunk 5]
```diff
@@ -68,6 +69,15 @@ def from_qualified_name(cls, qualified_name: str) -> OpName:
     def from_op_overload(cls, op_overload: torch._ops.OpOverload) -> OpName:
         return cls.from_qualified_name(op_overload.name())
 
+    @classmethod
+    @_beartype.beartype
+    def from_builtin_function(
+        cls, builtin_function: types.BuiltinFunctionType
+    ) -> OpName:
+        op = builtin_function.__name__  # add, sub, etc.
```
Comment (thiagocrepaldi) [inline torch/onnx/_internal/fx/registration.py:77]: add docstring

**Review by thiagocrepaldi — COMMENTED** [review]
Awesome, having built-in ops seems a much more robust solution

**Comment by thiagocrepaldi** [comment]
NIT: did you really mean "sympy" in the PR title? or was it "built-in"?

**Review by thiagocrepaldi — APPROVED** [review]

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py [hunk 6]
```diff
@@ -147,9 +148,9 @@ def _retrieve_or_adapt_input_to_graph_set(
         # onnx-script auto wraps python number with op.Constants,
         # so we don't need to specifically process them.
         with onnxscript.evaluator.default_as(tracer):
-            output = onnxscript.opset18.Concat(*sequence_mixed_elements, axis=0)
-        output.dtype = torch.int64
-        output.shape = [len(sequence_mixed_elements)]
+            output = onnxscript.opset18.Concat(*sequence_mixed_elements, axis=0)  # type: ignore[type-var]
```
Comment (justinchuby) [inline torch/onnx/_internal/fx/fx_onnx_interpreter.py:151]: We should alias the concat op in torch lib in the near future so this call is opset version agnostic

**Pull request closed without merging**